addons/base_vat/models/res_partner.py

```diff
@@ -707,40 +707,44 @@ def format_vat_sm(self, vat):
         stdnum_vat_format = stdnum.util.get_cc_module('sm', 'vat').compact
         return stdnum_vat_format('SM' + vat)[2:]
 
-    def _fix_vat_number(self, vat, country_id):
-        code = self.env['res.country'].browse(country_id).code if country_id else False
-        vat_country, vat_number = self._split_vat(vat)
-        if code and code.lower() != vat_country:
-            return vat
-        stdnum_vat_fix_func = getattr(stdnum.util.get_cc_module(vat_country, 'vat'), 'compact', None)
-        #If any localization module need to define vat fix method for it's country then we give first priority to it.
-        format_func_name = 'format_vat_' + vat_country
+    @api.model
+    def _format_vat_number(self, country_code, vat):
+        """ Low-level method directly calling stdnum or our own specific method returning the formatted VAT. """
+        stdnum_vat_fix_func = getattr(stdnum.util.get_cc_module(country_code, 'vat'), 'compact', None)
+        # If any localization module needs to define vat fix method for its country then we give first priority to it.
+        format_func_name = 'format_vat_' + country_code.lower()
         format_func = getattr(self, format_func_name, None) or stdnum_vat_fix_func
         if format_func:
-            vat_number = format_func(vat_number)
-        return vat_country.upper() + vat_number
+            vat = format_func(vat)
+        return vat
 
     @api.model
     def _convert_hu_local_to_eu_vat(self, local_vat):
-        if self.__check_tin_hu_companies_re.match(local_vat):
+        if self._check_tin_hu_companies_re.match(local_vat):
             return f'HU{local_vat[:8]}'
         return False
 
+    def _get_vat_required_valid(self, company=None):
+        # OVERRIDE
+        # If VIES validation does not apply to this partner (e.g. they
+        # are in the same country as the partner), then skip.
+        vat_required_valid = super()._get_vat_required_valid(company=company)
+        eu_countries = self.env.ref('base.europe').country_ids
+        if (
+            company and self.with_company(company).perform_vies_validation
+            and (company.country_id in eu_countries or self.with_company(company)._eu_fiscalpos_exists())
```

Comment:
Avoid using `ref()` for country groups
```suggestion
            and (
                'EU' in company.country_id.country_group_codes
                or self.with_company(company)._eu_fiscalpos_exists()
            )
```